### PR TITLE
Add colorspace functions: rgb2ycbcr, bgr2ycbcr, ycbcr2rgb and ycbcr2bgr

### PR DESCRIPTION
**Pull request by xinntao**
Add colorspace functions: rgb2ycbcr, bgr2ycbcr, ycbcr2rgb and ycbcr2bgr

These functions produce almost the same results as Matlab's rgb2ycbcr, ycbcr2rgb function.

### CODE REVIEW

File: mmcv/image/colorspace.py [hunk 1]
```diff
@@ -73,13 +74,124 @@ def gray2rgb(img):
...
+            1. np.uint8 type with range [0, 255];
+            2. np.float32 type with range [0, 1].
+        return_y (bool): Whether to only return Y channel. Default: False.
+
+    Returns:
+        ndarray: The converted YCbCr image. The output image has the same type
+            and range as input image.
+    """
+    img_type = img.dtype
+    img = img.copy().astype(np.float32)
```
Comment (yhcao6) [inline mmcv/image/colorspace.py:100]: copy() is redundant.

File: tests/test_image/test_colorspace.py [hunk 2]
```diff
@@ -89,6 +89,213 @@ def test_bgr2hsv():
...
+def test_rgb2ycbcr():
+    with pytest.raises(TypeError):
+        # The img type should be np.float32 or np.uint8
+        in_img = np.random.rand(10, 10, 3).astype(np.uint64)
+        mmcv.rgb2ycbcr(in_img)
+
+    # float32
+    in_img = np.random.rand(10, 10, 3).astype(np.float32)
+    out_img = mmcv.rgb2ycbcr(in_img)
+    computed_ycbcr = np.empty_like(in_img, dtype=in_img.dtype)
```
Comment (yhcao6) [inline tests/test_image/test_colorspace.py:101]: `dtype=in_img.dtype` is redundant.

File: tests/test_image/test_colorspace.py [hunk 3]
```diff
@@ -89,6 +89,213 @@ def test_bgr2hsv():
...
+    for i in range(in_img.shape[0]):
+        for j in range(in_img.shape[1]):
+            r = in_img[i, j, 0]
+            g = in_img[i, j, 1]
+            b = in_img[i, j, 2]
```
Comment (yhcao6) [inline tests/test_image/test_colorspace.py:106]: r, g, b = in_img[i, j]

File: mmcv/image/colorspace.py (hunk 1 above)
Comment (xinntao) [reply mmcv/image/colorspace.py:100]: If we do not use copy, this function will modify the original img (such  range). If we want to use the original img later, this modification will cause problems.

**Comment by xinntao** [comment]
@yhcao6  have updated~

File: mmcv/image/colorspace.py [hunk 4]
```diff
@@ -73,13 +74,124 @@ def gray2rgb(img):
...
     """
     img = img[..., None] if img.ndim == 2 else img
     out_img = cv2.cvtColor(img, cv2.COLOR_GRAY2RGB)
     return out_img
 
 
+def rgb2ycbcr(img, return_y=False):
+    """Convert a RGB image to YCbCr image.
+
+    This function produces the same results as Matlab's rgb2ycbcr function.
```
Comment (hellock) [inline mmcv/image/colorspace.py:87]: May also address the difference from opencv.

File: mmcv/image/colorspace.py [hunk 5]
```diff
@@ -73,13 +74,124 @@ def gray2rgb(img):
...
+    """
+    img_type = img.dtype
+    img = img.copy().astype(np.float32)
+    if img_type == np.float32:
+        pass
+    elif img_type == np.uint8:
+        img /= 255.
+    else:
+        raise TypeError('The img type should be np.float32 or np.uint8, '
+                        f'but got {img_type}')
```
Comment (hellock) [inline mmcv/image/colorspace.py:107]: The above logic seems to be shared by multiple methods. You may create a util method.

File: mmcv/image/colorspace.py [hunk 6]
```diff
@@ -73,13 +74,124 @@ def gray2rgb(img):
...
+            1. np.uint8 type with range [0, 255];
+            2. np.float32 type with range [0, 1].
+        return_y (bool): Whether to only return Y channel. Default: False.
+
+    Returns:
+        ndarray: The converted YCbCr image. The output image has the same type
+            and range as input image.
+    """
+    img = bgr2rgb(img)
+    return rgb2ycbcr(img, return_y=return_y)
```
Comment (hellock) [inline mmcv/image/colorspace.py:138]: Using two converting methods may introduce additional overhead.

File: mmcv/image/colorspace.py [hunk 7]
```diff
@@ -73,13 +74,124 @@ def gray2rgb(img):
...
     Returns:
-        ndarray: The converted BGR image.
+        ndarray: The converted RGB image.
     """
     img = img[..., None] if img.ndim == 2 else img
     out_img = cv2.cvtColor(img, cv2.COLOR_GRAY2RGB)
     return out_img
 
 
+def rgb2ycbcr(img, return_y=False):
```
Comment (hellock) [inline mmcv/image/colorspace.py:84]: `return_y` -> `y_only`

File: mmcv/image/colorspace.py (hunk 1 above)
Comment (yhcao6) [reply mmcv/image/colorspace.py:100]: I think `astype` will create a copy of `img`. Is it not?

Comment (xinntao) [reply mmcv/image/colorspace.py:100]: Yes, you are right~ I have modified it

**Comment by xinntao** [comment]
@hellock  have updated~

**Pull request merged**